src/fluree/db/json_ld/shacl.cljc

```diff
@@ -1152,12 +1152,18 @@
   (let [p-ns (property-ns-code flake)]
     (= p-ns shacl-ns-code)))
 
-(defn extract-shapes
+(defn hydrate-shape-cache!
   [db]
   (go-try
     (let [new-shapes (<? (rebuild-shapes db))]
-      (reset-shape-cache! db new-shapes) ;; TODO - there is really no reason this needs to be an atom any longer
-      (vals new-shapes))))
+      (reset-shape-cache! db new-shapes)
```

Comment:
I think you should change the `reset-shape-cache!` function to return the `db` it was passed in.